web/source/dom/domEventHandlers.ts

```diff
@@ -293,16 +270,14 @@ namespace com.keyman.dom {
      * Description          Stores the last active element's keyboard settings.  Should be called
      *                      whenever a KMW-enabled page element loses control.
      */
-    _BlurKeyboardSettings(PInternalName?: string, PLgCode?: string) {
+    _BlurKeyboardSettings(lastElem: HTMLElement, PInternalName?: string, PLgCode?: string) {
       var keyboardID = this.keyman.core.activeKeyboard ? this.keyman.core.activeKeyboard.id : '';
       var langCode = this.keyman.keyboardManager.getActiveLanguage();
       
       if(PInternalName !== undefined && PLgCode !== undefined) {
         keyboardID = PInternalName;
         langCode = PLgCode;
       }
-      
-      var lastElem = DOMEventHandlers.states.lastActiveElement;
 
       if(lastElem && lastElem._kmwAttachment.keyboard != null) {
         lastElem._kmwAttachment.keyboard = keyboardID;
```

Comment:
Won't this now modify the parameter `lastElem`?

Reply:
To directly answer your question, yes, it does modify (a property of) the parameter.

That said, that parameter references a KMW-managed property of a persistent, global page element.  The parameter is passed to this method for _exactly_ this reason - to perform related updates to the element's KMW-managed properties.